tooling/greybox_fuzzer/src/mutation/array.rs

```diff
@@ -0,0 +1,201 @@
+use std::cmp::min;
+
+use noirc_abi::input_parser::InputValue;
+use rand::Rng;
+use rand_xorshift::XorShiftRng;
+
+use crate::mutation::configurations::{SpliceMutation, BASIC_SPLICE_MUTATION_CONFIGURATION};
+
+use super::configurations::{
+    SpliceCandidate, StructuralMutation, BASIC_SPLICE_CANDIDATE_PRIORITIZATION_CONFIGURATION,
+    BASIC_VECTOR_STRUCTURE_MUTATION_CONFIGURATION,
+};
+struct ArrayMutator<'a> {
+    prng: &'a mut XorShiftRng,
+}
+
+impl<'a> ArrayMutator<'a> {
+    pub fn new(prng: &'a mut XorShiftRng) -> Self {
+        Self { prng }
+    }
+
+    /// Create a spliced version of 2 buffers, where each element in the result is at the same index as in the original ones
+    fn structured_splice(
+        &mut self,
+        first_buffer: &[InputValue],
+        second_buffer: &[InputValue],
+    ) -> Vec<InputValue> {
```

Comment:
I wonder if this one and the one in string could be done by calling a generic helper function with `T` being `InputValue` or `u8` (the logic seems to be the same)

Reply:
Same goes for `swap` and `duplicate_chunk`